support/util/networking.go

```diff
@@ -121,3 +130,43 @@ func AllowedCIDRBlocks(hcp *hyperv1.HostedControlPlane) []hyperv1.CIDRBlock {
 	}
 	return nil
 }
+
+func SetAdvertiseAddresses(hcp *hyperv1.HostedControlPlane, defaults *DefaultAdvIps) []string {
+	addresses := make([]string, 0)
+
+	// User defined AdvertiseAddress/AdvertiseAddresses
+	if hcpAdvAddresses := AdvertiseAddresses(hcp); hcpAdvAddresses != nil {
+		fmt.Println("AdvertiseAddresses found:", hcpAdvAddresses)
+		addresses = hcpAdvAddresses
+	}
+
+	if hcpAdvAddress := AdvertiseAddress(hcp); hcpAdvAddress != nil {
+		fmt.Println("AdvertiseAddress found:", hcpAdvAddress)
+		addresses = append(addresses, *hcpAdvAddress)
+	}
+
+	// If there is not any defined AdvertiseAddress we should default them.
+	if len(addresses) == 0 {
+		fmt.Println("AdvertiseAddress or AdvertiseAddresses not found in hcp, using default values")
+		addresses = GetIpsForSvcNetsWithDefaults(hcp.Spec.Networking.ServiceNetwork, defaults)
+	}
+
+	// Clean duplicates from the slice
+	finalList := RemoveDuplicatesFromList(addresses)
```

Comment:
why not just use a map here for `addresses`, and then return the keys.

Reply:
Just to avoid human errors including repeated CIDRs in the spec

Reply:
I ended commenting that line, I have that prepared to handle the situation when the issue gets solved.